### ISSUE TITLE
checking rightclick event is not working if dye has named.

### ISSUE THREAD

**Issue by mckite** [SkriptLang/Skript#2651]
### Description

checking rightclick event is not correctly working if dye has named.

### Steps to Reproduce

compile this code:

on rightclick holding rose red: 
	send "%event-item%"
on rightclick holding any dye: 
	send "%event-item%"

and rightclick with rose red named.

### Expected Behavior
send message "red dye named ~~~"

### Errors / Screenshots

When rightclick with red dye which not named works well
but, when rightclick with dye which named is not work. (No respons without lag)

### Server Information
* **Server version/platform:**     Paper version git-Paper-1618 (MC: 1.12.2)
* **Skript version:**                2.3.7

### Additional Context

on rightclick holding rose red: 
	send "%event-item%"
on rightclick holding any dye: 
	send "%event-item%"
on rightclick:
       if type of tool is any dye:
            send "%event-item%"

**Comment by ShaneBeee** [comment]
I can replicate this on Skript 2.4-beta10

**Comment by bensku** [comment]
Appears to be fixed by #2834.

**Comment by ShaneBeee** [comment]
@bensku 
## Testing results:
### code:
```vb
on rightclick holding rose red:
    send "ROSE RED EVENT: %event-item% &r:: %player's tool%"
on rightclick holding any dye:
    send "DYE EVENT: %event-item% &r:: %player's tool%"
on rightclick:
    if type of tool is any dye:
        send "EVENT: %event-item% &r:: %player's tool%"
```
### 2.3.7 and 2.4.1 on 1.12.2:
- using the above code, the events which include the item do not appear to fire when the item is named. The regular click event does fire.
<img width="635" alt="Screen Shot 2020-02-29 at 12 26 58 PM" src="https://user-images.githubusercontent.com/20008482/75614555-ca3e0580-5aee-11ea-8400-5abc6d0ef2f8.png">


### Using build from #2834 
- Using the above code, the events all correctly fire when using an unnamed item or a named item.
(**NOTE:** new issue arose where `%event-item%` is returning the clicked block rather than the player's tool as with previous versions.)
<img width="635" alt="Screen Shot 2020-02-29 at 12 29 26 PM" src="https://user-images.githubusercontent.com/20008482/75614597-45072080-5aef-11ea-8dc0-54691ddedb34.png">

**Comment by ShaneBeee** [comment]
Im marking this as completed and closing it.
Thru my testing it seems to have been fixed in 2.5-alpha2
